backend/app/services/share/knowledge_share_service.py

```
@@ -128,6 +158,12 @@ def _get_resource(
             )
             role = get_effective_role_in_group(db, user_id, kb.namespace)
             if role is not None:
+                # RestrictedAnalyst is not allowed to access knowledge base details
+                if role == GroupRole.RestrictedAnalyst:
+                    logger.warning(
+                        f"[_get_resource] User has team role but is RestrictedAnalyst"
```

Ruff (0.15.6)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)